src/utils/switch.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Transpose a matrix
+ * @param {Array} mat
+ * @returns {Array} ret
+ * @private
+ */
+export function _switch (mat) {
```

Comment:
Makes a lot of sense de-duplicating this helper function :smile:

Reply:
I think I introduced the re-use of this function and so felt responsible to not duplicate it.